### ISSUE TITLE
Too many callbacks

### ISSUE THREAD

**Issue by LPGhatguy** [LPGhatguy/luajit-request#3]
I got a request about this via email, stating:

"I'm using luajit-request in order to get a lot of pages, so eventually it ends up failing because of too many callbacks (on luajit-request/init.lua:148).
Would you know what I could do to avoid doing so many callbakcs, or to free them when I done using them?"

**Comment by shakesoda** [comment]
yeah, this is one of those horribly sneaky FFI gotchas. they _can_ be freed, however, according to [the docs](http://luajit.org/ext_ffi_api.html#callback)

**Comment by LPGhatguy** [comment]
If anyone wants to take a look at this, especially someone who uses the library, I'd really appreciate it.

**Comment by LPGhatguy** [comment]
I believe I'll have this fixed in the next push up!

**Comment by LPGhatguy** [comment]
Done in 2.3.0!

**Comment by Fallstar** [comment]
Hi, just tried it, still does too many callbacks. :/
I'll try to make a test suite when I'm done with my other projects.

**Comment by LPGhatguy** [comment]
That'd be much appreciated!
